cdap-ui/app/features/dashboard/controllers/widget.js

```diff
@@ -14,17 +14,39 @@ angular.module(PKG.name+'.feature.dashboard')
       this.dataSrc = null;
       this.isLive = false;
     }
+
+    // 'ns.default.app.foo' -> {'ns': 'default', 'app': 'foo'}
+    function contextToTags(context) {
+      parts = context.split('.');
+      if (parts.length % 2 != 0) {
+        throw "Metrics context has uneven number of parts: " + this.metric.context
+      }
+      tags = {};
+      for (var i = 0; i < parts.length; i+=2) {
+        tags[parts[i]] = parts[i + 1]
+      }
+      return tags;
+    }
+
+    function constructQuery(queryId, tags, metrics) {
+      timeRange = {'start': 'now-60s', 'end': 'now'};
+      queryObj = {tags: tags, metrics: metrics, groupBy: [], timeRange: timeRange};
+      retObj = {};
+      retObj[queryId] = queryObj;
+      return retObj;
+    }
+
+    // The queryId value does not matter, as long as we are using the same value in the request
+    // as in parsing the response.
+    var queryId = "qid";
     Widget.prototype.fetchData = function(scope) {
       if (!this.dataSrc) {
         this.dataSrc = new MyDataSource(scope);
       }
       this.dataSrc.request({
-        _cdapPath: '/metrics/query' +
-          '?context=' + encodeURIComponent(this.metric.context) +
-          '&metric=' + encodeURIComponent(this.metric.name) +
-          '&start=now-60s&end=now',
-
-        method: 'POST'
+        _cdapPath: '/metrics/query',
```

Comment:
So either metrics parameter has to be in query params, or send it as the body of the request. I am changing it to use the body of the request.